sources/api/apiserver/src/server/mod.rs

```diff
@@ -453,6 +489,43 @@ async fn get_affected_services(
 
 /// Get all settings that have setting-generator metadata
 async fn get_setting_generators(data: web::Data<SharedData>) -> Result<MetadataResponse> {
+    let datastore = data.ds.read().ok().context(error::DataStoreLockSnafu)?;
+    let metadata_for_keys =
+        controller::get_metadata_for_all_data_keys(&*datastore, "setting-generator")?;
+    let mut resp: HashMap<String, Value> = HashMap::new();
+
+    for (key, value) in metadata_for_keys.iter() {
+        match value {
```

Comment:
Mentioned in another comment, but if we allowed `SettingsGenerator` to successfully deserialize from a `String` or an `Object`, we could remove a branch and make this much simpler -- just parse the returned metadata into a `SettingsGenerator`